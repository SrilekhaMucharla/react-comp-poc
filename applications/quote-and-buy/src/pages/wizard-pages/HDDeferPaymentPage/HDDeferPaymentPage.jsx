import React from 'react';
import PropTypes from 'prop-types';
import {
    Container, Row, Col
} from 'react-bootstrap';
import { HDInfoCard } from 'hastings-components';
import {
    AnalyticsHDToggleButtonGroup as HDToggleButtonGroup,
    AnalyticsHDButton as HDButton
} from '../../../web-analytics/index';
import TipCirclePurple from '../../../assets/images/icons/tip_circle_purple.svg';
import BackNavigation from '../../Controls/BackNavigation/BackNavigation';
import * as messages from './HDDeferPaymentPage.messages';
import getDeferText from './helpers';
import './HDDeferPaymentPage.scss';

export default function HDDeferPaymentPage({ payments }) {
    return (
        <Container className="container-defer-payment">
            <Row>
                <Col xs={12} md={10}>
                    <BackNavigation />
                    <h1 className="heading-primary u-margin-bottom-mid">{messages.mainHeading}</h1>
                    <HDToggleButtonGroup
                        webAnalyticsEvent={{ event_action: messages.mainHeading }}
                        id="pay-all-button-group"
                        className="btn-group-defer u-margin-bottom-mid"
                        availableValues={
                            [
                                {
                                    value: 'payFull',
                                    content:
    <>
        <span>{messages.payFull}</span>
        <span className="btn-secondary-description">
            {`(Â£${payments.reduce((prev, curr) => prev + curr.amount, 0)} for all cars)`}
        </span>
    </>
                                },
                                {
                                    value: 'payDefer',
                                    content:
    <>
        <span>{messages.payDefer}</span>
        <span data-testid="text-defer" className="btn-secondary-description">
            {getDeferText(payments)}
        </span>
    </>
                                }
                            ]
                        } />
                    <HDInfoCard className="u-margin-bottom-big">
                        <img src={TipCirclePurple} alt="tip icon" />
                        <p style={{ margin: 0 }}>
                            {messages.infoCardContent}
                        </p>
                    </HDInfoCard>
                    <HDButton
                        webAnalyticsEvent={{ event_action: messages.continueRedirect }}
                        id="continue-to-summary-button"
                        variant="btnprimary"
                        label={null}
                    >
                        {messages.continueToSummary}
                    </HDButton>
                </Col>
            </Row>
        </Container>
    );
}

HDDeferPaymentPage.propTypes = {
    payments: PropTypes.arrayOf(PropTypes.shape({
        dueDate: PropTypes.string,
        amount: PropTypes.number
    })),
};

HDDeferPaymentPage.defaultProps = {
    payments: [{
        dueDate: 'today',
        amount: 458.50
    }, {
        dueDate: '15/12/2020',
        amount: 301.40
    }, {
        dueDate: '02/02/2021',
        amount: 237.16
    }]
};
