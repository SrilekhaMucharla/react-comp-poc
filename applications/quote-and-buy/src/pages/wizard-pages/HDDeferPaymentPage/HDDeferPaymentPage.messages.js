export const mainHeading = 'What would you like to do about your payments?';
export const infoCardContent = `If you choose to pay some of these later, we'll take payment automatically on the date it's due.
But don't worry, we'll always send you  a reminder nearer the time.`;
export const payFull = 'Pay all today';
export const payDefer = 'Pay when each policy starts';
export const continueToSummary = 'Continue to summary';
export const continueRedirect = 'Continue - Redirecting from: HastingsMCDeferPayment';
