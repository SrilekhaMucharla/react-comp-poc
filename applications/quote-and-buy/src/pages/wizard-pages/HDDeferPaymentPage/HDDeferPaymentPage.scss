@import "vars";

// most of these styles are temporary
// once HDToggleButtonGroup is refactored, they will be replaced
.container-defer-payment {
    .container {
        &-defer-payment {
            &::before {
                content: "";
                position: fixed;
                width: 100%;
                height: 100%;
                top: 0;
                left: 0;
                background-color: $color-blue-dark-2;
            }
        }

        .btn-group {
            &-defer {
                > .btn-group {
                    display: flex;
                    gap: 30px;

                    @include atTablet {
                        flex-direction: column;
                        gap: 20px;
                    }

                    > label {
                        display: flex;
                        flex-direction: column;
                        justify-content: center;
                        min-height: 70px;
                        height: auto !important;
                        color: $color-white;
                        background-color: $color-blue-dark-1;
                        border-radius: 6px !important;
                        width: 100%;

                        &:hover {
                            border: 1px solid $color-blue-light !important;
                        }
                    }

                    > .btn {
                        margin: 0;

                        &.active {
                            border: 1px solid $color-blue-light !important;
                        }
                    }
                }
            }
        }

        [class^="col-"] {
            margin: 0 auto;
        }

        .heading-primary {
            font-family: TTNormsPro-Bold;
            font-size: 34px;
            line-height: 1.2;
            color: $color-white;
            margin-top: 50px;

            @include atTablet {
                font-size: 25px;
                margin-top: 30px;
            }
        }

        .btn-secondary-description {
            font-family: TTNormsPro;
            font-size: 12px;
            font-weight: normal;
        }

        .u-margin-bottom-mid {
            margin-bottom: 30px;
        }

        .u-margin-bottom-big {
            margin-bottom: 50px;

            @include atTablet {
                margin-bottom: 40px;
            }
        }
    }
}
