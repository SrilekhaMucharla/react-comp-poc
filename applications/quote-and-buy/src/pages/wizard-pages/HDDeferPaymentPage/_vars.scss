$color-blue-light: #0085ff; 
$color-blue-dark-1: #01355a;
$color-blue-dark-2: #011831;
$color-white: #fff;

$bootstrap-sm-width: 576px;
$bootstrap-md-width: 768px;

@mixin atMobile {
  @media (max-width: #{$bootstrap-sm-width}) {
    @content;
  }
}

@mixin atTablet {
  @media (max-width: #{$bootstrap-md-width}) {
    @content;
  }
}