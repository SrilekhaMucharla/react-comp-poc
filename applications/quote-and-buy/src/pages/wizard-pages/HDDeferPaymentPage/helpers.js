const mapPaymentToText = (payment, i) => {
    return `Â£${payment.amount} ${i !== 0 ? `on ${payment.dueDate}` : payment.dueDate}`;
};

const getDeferText = (payments) => {
    const lastIndex = payments.length - 1;

    return [
        payments.slice(0, lastIndex).map(mapPaymentToText).join(', '),
        mapPaymentToText(payments[lastIndex])
    ].join(' and ');
};

export default getDeferText;
