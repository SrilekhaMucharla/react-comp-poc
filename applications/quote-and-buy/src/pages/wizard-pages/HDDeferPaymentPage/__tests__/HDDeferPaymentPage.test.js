import React from 'react';
import Enzyme from 'enzyme';
import HDDeferPaymentPage from '../HDDeferPaymentPage';

const defaultProps = { ...HDDeferPaymentPage.defaultProps };

const setup = (props = defaultProps) => { return Enzyme.mount(<HDDeferPaymentPage {...props} />); };
const findByTestAttribute = (wrapper, selector) => wrapper.find(`[data-testid='${selector}']`);

describe('<HDDeferPaymentPage />', () => {
    it('should render page with default props', () => {
        const tree = setup();
        expect(tree).toMatchSnapshot();
    });

    it('should display correct defer message', () => {
        const props = [{
            dueDate: 'today',
            amount: 458.50
        }, {
            dueDate: '15/12/2020',
            amount: 301.40
        }, {
            dueDate: '02/02/2021',
            amount: 237.16
        }, {
            dueDate: '02/05/2021',
            amount: 500.00
        }];

        const expectedMessage = '£458.5 today, £301.4 on 15/12/2020, £237.16 on 02/02/2021 and £500 on 02/05/2021';
        const component = setup({ payments: props });

        const actualMessage = findByTestAttribute(component, 'text-defer').text();

        expect(actualMessage).toEqual(expectedMessage);
    });
});
