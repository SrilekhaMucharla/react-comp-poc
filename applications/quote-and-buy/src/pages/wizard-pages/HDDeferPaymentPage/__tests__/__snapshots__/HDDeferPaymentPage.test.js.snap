// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDDeferPaymentPage /> should render page with default props 1`] = `
<HDDeferPaymentPage
  payments={
    Array [
      Object {
        "amount": 458.5,
        "dueDate": "today",
      },
      Object {
        "amount": 301.4,
        "dueDate": "15/12/2020",
      },
      Object {
        "amount": 237.16,
        "dueDate": "02/02/2021",
      },
    ]
  }
>
  <Container
    className="container-defer-payment"
    fluid={false}
  >
    <div
      className="container-defer-payment container"
    >
      <Row
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            md={10}
            xs={12}
          >
            <div
              className="col-md-10 col-12"
            >
              <BackNavigation
                className=""
                onClick={null}
                onKeyPress={null}
              >
                <ForwardRef
                  className="go-back "
                  id="backNavMain"
                  label="Go back"
                  name="go-back"
                  onBeforeOpen={null}
                  onBlur={null}
                  onCancel={null}
                  onChange={null}
                  onClick={[Function]}
                  onConfirm={null}
                  onDelete={null}
                  onEdit={null}
                  onFocus={null}
                  onKeyPress={[Function]}
                  onSelect={null}
                  variant="default"
                  webAnalyticsEvent={
                    Object {
                      "event_action": "Go back",
                    }
                  }
                >
                  <HDButtonRefactor
                    className="go-back "
                    disabled={false}
                    fullWidth={false}
                    id="backNavMain"
                    label="Go back"
                    name="go-back"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    size={null}
                    type="button"
                    variant="default"
                  >
                    <Button
                      active={false}
                      block={false}
                      className="hd-btn hd-btn-default go-back "
                      disabled={false}
                      id="backNavMain"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      size={null}
                      type="button"
                      variant="default"
                    >
                      <button
                        className="hd-btn hd-btn-default go-back  btn btn-default"
                        disabled={false}
                        id="backNavMain"
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        type="button"
                      >
                        <span
                          className="back-icon"
                        >
                          <_default>
                            <svg
                              height="14"
                              viewBox="0 0 8 14"
                              width="8"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M0.225564 6.49312C0.0845862 6.6422 -2.91944e-07 6.8211 -2.84124e-07 7C-2.75001e-07 7.20872 0.0845862 7.38761 0.225564 7.50688L5.69549 13.2913C5.83647 13.4404 6.00564 13.5 6.17481 13.5C6.37218 13.5 6.54135 13.4404 6.65414 13.2913L7.27444 12.6353C7.41541 12.5161 7.5 12.3372 7.5 12.1284C7.5 11.9495 7.44361 11.7706 7.30263 11.6216L2.93233 7L7.30263 2.37844C7.44361 2.25917 7.5 2.08028 7.5 1.87156C7.5 1.69266 7.41541 1.51376 7.27444 1.36468L6.65414 0.708715C6.54135 0.58945 6.37218 0.5 6.17481 0.5C6.00564 0.5 5.83647 0.58945 5.69549 0.708715L0.225564 6.49312Z"
                                fill="#0069CC"
                              />
                            </svg>
                          </_default>
                        </span>
                        Go back
                      </button>
                    </Button>
                  </HDButtonRefactor>
                </ForwardRef>
              </BackNavigation>
              <h1
                className="heading-primary u-margin-bottom-mid"
              >
                What would you like to do about your payments?
              </h1>
              <ForwardRef
                availableValues={
                  Array [
                    Object {
                      "content": <React.Fragment>
                        <span>
                          Pay all today
                        </span>
                        <span
                          className="btn-secondary-description"
                        >
                          (£997.06 for all cars)
                        </span>
                      </React.Fragment>,
                      "value": "payFull",
                    },
                    Object {
                      "content": <React.Fragment>
                        <span>
                          Pay when each policy starts
                        </span>
                        <span
                          className="btn-secondary-description"
                          data-testid="text-defer"
                        >
                          £458.5 today, £301.4 on 15/12/2020 and £237.16 on 02/02/2021
                        </span>
                      </React.Fragment>,
                      "value": "payDefer",
                    },
                  ]
                }
                className="btn-group-defer u-margin-bottom-mid"
                id="pay-all-button-group"
                onBeforeOpen={null}
                onBlur={null}
                onCancel={null}
                onChange={null}
                onClick={null}
                onConfirm={null}
                onDelete={null}
                onEdit={null}
                onFocus={null}
                onSelect={null}
                webAnalyticsEvent={
                  Object {
                    "event_action": "What would you like to do about your payments?",
                  }
                }
              >
                <HDToggleButtonGroup
                  availableValues={
                    Array [
                      Object {
                        "content": <React.Fragment>
                          <span>
                            Pay all today
                          </span>
                          <span
                            className="btn-secondary-description"
                          >
                            (£997.06 for all cars)
                          </span>
                        </React.Fragment>,
                        "value": "payFull",
                      },
                      Object {
                        "content": <React.Fragment>
                          <span>
                            Pay when each policy starts
                          </span>
                          <span
                            className="btn-secondary-description"
                            data-testid="text-defer"
                          >
                            £458.5 today, £301.4 on 15/12/2020 and £237.16 on 02/02/2021
                          </span>
                        </React.Fragment>,
                        "value": "payDefer",
                      },
                    ]
                  }
                  btnClassName=""
                  btnGroupClassName=""
                  className="btn-group-defer u-margin-bottom-mid"
                  customClassName=""
                  data={null}
                  disabled={false}
                  doReset={false}
                  id="pay-all-button-group"
                  label={null}
                  name="toggle-button-name"
                  onChange={[Function]}
                  path="path.to.toggle-button"
                  type="radio"
                  value=""
                >
                  <div
                    className=" btn-group-defer u-margin-bottom-mid"
                  >
                    <ForwardRef
                      className=""
                      name="toggle-button-name"
                      type="radio"
                      value=""
                      vertical={false}
                    >
                      <ButtonGroup
                        className=""
                        role="group"
                        toggle={true}
                        vertical={false}
                      >
                        <div
                          className="btn-group btn-group-toggle"
                          role="group"
                        >
                          <ToggleButton
                            checked={false}
                            className=""
                            disabled={false}
                            key=".$payFull"
                            name="toggle-button-name"
                            onChange={[Function]}
                            type="radio"
                            value="payFull"
                            variant="default"
                          >
                            <Button
                              active={false}
                              as="label"
                              className=""
                              disabled={false}
                              variant="default"
                            >
                              <label
                                className="btn btn-default"
                                disabled={false}
                              >
                                <input
                                  autoComplete="off"
                                  checked={false}
                                  disabled={false}
                                  name="toggle-button-name"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  type="radio"
                                  value="payFull"
                                />
                                <span>
                                  Pay all today
                                </span>
                                <span
                                  className="btn-secondary-description"
                                >
                                  (£997.06 for all cars)
                                </span>
                              </label>
                            </Button>
                          </ToggleButton>
                          <ToggleButton
                            checked={false}
                            className=""
                            disabled={false}
                            key=".$payDefer"
                            name="toggle-button-name"
                            onChange={[Function]}
                            type="radio"
                            value="payDefer"
                            variant="default"
                          >
                            <Button
                              active={false}
                              as="label"
                              className=""
                              disabled={false}
                              variant="default"
                            >
                              <label
                                className="btn btn-default"
                                disabled={false}
                              >
                                <input
                                  autoComplete="off"
                                  checked={false}
                                  disabled={false}
                                  name="toggle-button-name"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  type="radio"
                                  value="payDefer"
                                />
                                <span>
                                  Pay when each policy starts
                                </span>
                                <span
                                  className="btn-secondary-description"
                                  data-testid="text-defer"
                                >
                                  £458.5 today, £301.4 on 15/12/2020 and £237.16 on 02/02/2021
                                </span>
                              </label>
                            </Button>
                          </ToggleButton>
                        </div>
                      </ButtonGroup>
                    </ForwardRef>
                  </div>
                </HDToggleButtonGroup>
              </ForwardRef>
              <HDInfoCard
                className="u-margin-bottom-big"
                image={null}
                isRight={false}
                notes={null}
                paragraphs={Array []}
                title={null}
              >
                <styled.div
                  className="i-card-container left u-margin-bottom-big"
                >
                  <div
                    className="sc-iCoGMd hIYlRK i-card-container left u-margin-bottom-big"
                  >
                    <img
                      alt="tip icon"
                      src="tip_circle_purple.svg"
                    />
                    <p
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                      If you choose to pay some of these later, we'll take payment automatically on the date it's due.
But don't worry, we'll always send you  a reminder nearer the time.
                    </p>
                  </div>
                </styled.div>
              </HDInfoCard>
              <ForwardRef
                id="continue-to-summary-button"
                label={null}
                onBeforeOpen={null}
                onBlur={null}
                onCancel={null}
                onChange={null}
                onClick={null}
                onConfirm={null}
                onDelete={null}
                onEdit={null}
                onFocus={null}
                onSelect={null}
                variant="btnprimary"
                webAnalyticsEvent={
                  Object {
                    "event_action": "Continue - Redirecting from: HastingsMCDeferPayment",
                  }
                }
              >
                <HDButtonRefactor
                  className=""
                  disabled={false}
                  fullWidth={false}
                  id="continue-to-summary-button"
                  label={null}
                  onClick={[Function]}
                  onKeyPress={null}
                  size={null}
                  type="button"
                  variant="btnprimary"
                >
                  <Button
                    active={false}
                    block={false}
                    className="hd-btn "
                    disabled={false}
                    id="continue-to-summary-button"
                    onClick={[Function]}
                    onKeyPress={null}
                    size={null}
                    type="button"
                    variant="btnprimary"
                  >
                    <button
                      className="hd-btn  btn btn-btnprimary"
                      disabled={false}
                      id="continue-to-summary-button"
                      onClick={[Function]}
                      onKeyPress={null}
                      type="button"
                    >
                      Continue to summary
                    </button>
                  </Button>
                </HDButtonRefactor>
              </ForwardRef>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </Container>
</HDDeferPaymentPage>
`;
